component/src/main/java/org/wso2/extension/siddhi/map/text/sourcemapper/TextSourceMapper.java

```diff
@@ -20,237 +20,421 @@
+                            "'true' in the stream " + streamID + " of siddhi text input mapper.");
+                    isValidEvent.set(false);
+                }
+                int position = attributePositionMap.get(attributeKey);
+                if ((Attribute.Type.STRING != attributeTypeMap.get(attributeKey)) && (matchText != null)) {
+                    data[position] = attributeConverter.getPropertyValue(matchText.replaceAll(",", ""), attributeTypeMap
+                            .get(attributeKey));
+                } else {
+                    data[position] = matchText;
+                }
+            }
         }
+        eventList.add(event);
+        return eventList.toArray(new Event[0]);
+    }
 
-        public void setRegex(String regex) {
-            this.regex = regex;
+    private Event[] convertToDefaultEvents(String eventObject) {
+        AtomicBoolean isValidEvent = new AtomicBoolean();
+        isValidEvent.set(true);
+        List<Event> eventList = new ArrayList<>();
+        Event event = new Event(this.streamDefinition.getAttributeList().size());
+        Object[] data = event.getData();
+        String[] events = eventObject.split(ATTRIBUTE_SEPARATOR + endOfLine);
+        if ((events.length < attributeList.size()) && (failOnMissingAttribute)) {
+            log.error("Invalid format of event because some required attributes are missing in event " + eventObject
+                    + " while needed attributes are " + attributeList.toString() + " in the stream " + streamID +
+                    " of siddhi text input mapper.");
+            isValidEvent.set(false);
         }
-
-        public int getGroupIndex() {
-            return groupIndex;
+        for (String event1 : events) {
+            String[] eventObjects = event1.split(KEY_VALUE_SEPARATOR, 2);
+            //remove trim
+            if (eventObjects.length == 2) {
+                String key = eventObjects[0].trim();
+                String value = eventObjects[1].trim();
+                Integer position = attributePositionMap.get(key.trim());
+                //to get 1st mach only
+                if (position != null) {
+                    if (!assignedPositionsBitSet.get(position)) {
+                        try {
+                            Attribute.Type attributeType = attributeTypeMap.get(key.trim());
+                            if (attributeType != Attribute.Type.STRING) {
+                                data[position] = attributeConverter.getPropertyValue(value.replaceAll(","
+                                        , "").trim(), attributeType);
+                            } else {
+                                data[position] = value.trim().substring(STRING_ENCLOSING_ELEMENT.length(),
+                                        +value.length() - STRING_ENCLOSING_ELEMENT.length());
+                            }
+                        } catch (ClassCastException | NumberFormatException e) {
```

Comment:
What is the reason behind catching exceptions to do this validation rather than pre-filtering these using conditional statements.

Reply:
Attribute converted throws those exceptions for invalid inputs so need to catch them to filter invalid inputs.